File: app/src/main/java/com/sg/moviesindex/db/typeconverters/CastTypeConverter.java
package com.sg.moviesindex.db.typeconverters;

import androidx.room.TypeConverter;

import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.sg.moviesindex.model.tmdb.Cast;

import java.lang.reflect.Type;
import java.util.ArrayList;

public class CastTypeConverter {

    private static final Gson gson = new Gson();

    @TypeConverter
    public static ArrayList<Cast> gettingListFromString(String data) {
        if (data == null) {
            return new ArrayList<Cast>();
        }

        Type listType = new TypeToken<ArrayList<Cast>>() {
        }.getType();

        return gson.fromJson(data, listType);
    }

    @TypeConverter
    public static String gettingStringFromList(ArrayList<Cast> someObjects) {
        return gson.toJson(someObjects);
    }

}

File: app/src/main/java/com/sg/moviesindex/data/local/typeconverters/CastTypeConverter.kt
package com.sg.moviesindex.data.local.typeconverters

import androidx.room.TypeConverter
import com.sg.moviesindex.data.remote.Cast
import com.squareup.moshi.Moshi
import com.squareup.moshi.Types

class CastTypeConverter {
  private val moshi = Moshi.Builder().build()
  private val listType = Types.newParameterizedType(List::class.java, Cast::class.java)
  private val adapter = moshi.adapter<List<Cast>>(listType)

  @TypeConverter
  fun gettingListFromString(data: String?): List<Cast> {
    if (data == null) {
      return emptyList()
    }
    return adapter.fromJson(data) ?: emptyList()
  }

  @TypeConverter
  fun gettingStringFromList(someObjects: List<Cast>): String = adapter.toJson(someObjects)
}
